tools/reviewer/web/src/app/dashboard/reviews-panel/reviews-panel.component.ts

```diff
@@ -1,88 +1,95 @@
-    if (!this.username) {
-      // Get user from AuthService
-      this.authService.getUser().subscribe(user => {
-        if (user) {
-          const email = user.email;
-          this.username = email.split('@')[0];
-        }
-      });
-    }
-    this.getReviews();
+  ) {
+    this.getReviews(this.authService.userEmail);
   }
 
   // Get diff/reviews from Database
-  getReviews() {
+  getReviews(userEmail: string) {
     this.firebaseService.getDiffs().subscribe(
-      protoDiffs => {
+      diffs => {
         // Diffs are categorized in 4 different lists
-        this.diffs[Lists.NeedAttention] = [];
-        this.diffs[Lists.CcedOn] = [];
-        this.diffs[Lists.DraftReviews] = [];
-        this.diffs[Lists.SubmittedReviews] = [];
+        this.diffGroups[Lists.NeedAttention] = [];
+        this.diffGroups[Lists.CcedOn] = [];
+        this.diffGroups[Lists.DraftReviews] = [];
+        this.diffGroups[Lists.SubmittedReviews] = [];
 
         // Iterate over each object in res
         // and create Diff from proto and categorize
         // into a specific list.
-        for (const protoDiff of protoDiffs) {
-          const diff: Diff.AsObject = protoDiff.toObject();
-          if (diff.needAttentionOfList.includes(this.username)) {
+        for (const diff of diffs) {
+          // needAttentionOfList is a list of author and all reviewers,
+          // where attention is requested
+          const needAttentionOfList: string[] = diff.getReviewerList()
+            .filter(reviewer => reviewer.getNeedsattention())
+            .map(reviewer => reviewer.getEmail());
+          if (diff.getAuthor().getNeedsattention()) {
+            needAttentionOfList.concat(diff.getAuthor().getEmail());
+          }
+
+          if (needAttentionOfList.includes(userEmail)) {
             // Need attention of user
-            this.diffs[Lists.NeedAttention].push(diff);
-          } else if (diff.ccList.includes(this.username)) {
+            this.diffGroups[Lists.NeedAttention].push(diff);
+          } else if (diff.getCcList().includes(userEmail)) {
             // User is cc'ed on this
-            this.diffs[Lists.CcedOn].push(diff);
-          } else if (diff.author === this.username && diff.status === 0) {
-            // Draft Review
-            this.diffs[Lists.DraftReviews].push(diff);
-          } else if (diff.author === this.username && diff.status === 4) {
-            // Submitted Review
-            this.diffs[Lists.SubmittedReviews].push(diff);
+            this.diffGroups[Lists.CcedOn].push(diff);
+          } else if (diff.getAuthor().getEmail() === userEmail) {
+            switch (diff.getStatus()) {
+              case Diff.Status.REVIEW_NOT_STARTED:
+                // Draft Review
+                this.diffGroups[Lists.DraftReviews].push(diff);
+                break;
+              case Diff.Status.SUBMITTED:
+                // Submitted Review
+                this.diffGroups[Lists.SubmittedReviews].push(diff);
```

Comment:
`SubmittedReviews` should be first, so if it's already submitted, it should not go into the other lists.